modules/stochastic_tools/src/utils/GaussianProcess.C

```diff
@@ -147,12 +152,215 @@ GaussianProcess::standardizeData(RealEigenMatrix & data, bool keep_moments)
     _data_standardizer.computeSet(data);
   _data_standardizer.getStandardized(data);
 }
+                                    
+void
+GaussianProcess::logl(const RealEigenMatrix & out_vec, const RealEigenMatrix & x1, const RealEigenMatrix & x2, Real g, const RealEigenMatrix & theta, 
+          LogLResult & result, bool outer, bool tau2) {
+  int n = out_vec.rows();
+  RealEigenMatrix K(x1.rows(), x2.rows());
+
+  std::vector<Real> theta1(_num_tunable, 0.0);
+  theta1[0] = 1;
+  theta1[1] = theta(0,0);
+  theta1[2] = theta(0,1);
+
+  vecToMap(_tuning_data, _hyperparam_map, _hyperparam_vec_map, theta1);
+  _covariance_function->loadHyperParamMap(_hyperparam_map, _hyperparam_vec_map);
+
+  _covariance_function->computeCovarianceMatrix(_K, x1, x2, true);
+
+  RealEigenMatrix Mi = _K.llt().solve(RealEigenMatrix::Identity(_K.rows(), _K.cols()));
+  Real ldet = std::log(_K.determinant());
+
+  RealEigenMatrix diff = out_vec;
+  Real quadterm = (diff.transpose() * Mi * diff)(0,0);
+
+  Real logl_val;
+  if (outer) {
+      logl_val = (-n * 0.5) * std::log(quadterm) - 0.5 * ldet;
+  } else {
+      logl_val = -0.5 * quadterm - 0.5 * ldet;
+  }
```

Comment:
This is a regular GP so no need for this. We could create an MCMC tuner that is general enough that could be reused here and the TwoLayerGP.